Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -994,11 +1029,33 @@ extension NSManagedObjectContext: MessageDatabaseSession {
         message.latestReactions = message.latestReactions.filter { $0 != reaction.id }
         message.ownReactions = message.ownReactions.filter { $0 != reaction.id }
 
-        guard let reactionScore = message.reactionScores.removeValue(forKey: type.rawValue), reactionScore > 1 else {
+        guard let reactionScore = message.reactionScores[type.rawValue] else {
             return reaction
         }
 
-        message.reactionScores[type.rawValue] = max(reactionScore - Int(reaction.score), 0)
+        let newScore = max(reactionScore - Int(reaction.score), 0)
+        message.reactionScores[type.rawValue] = newScore
+        message.reactionCounts[type.rawValue]? -= 1
+
+        let reactionGroup = message.reactionGroups.first(where: { $0.type == type.rawValue })
+        reactionGroup?.sumScores = Int64(newScore)
+        reactionGroup?.count -= 1
+        
+        let scoreIsZero = newScore == 0
+        let countIsZero = message.reactionCounts[type.rawValue] == 0
+
+        if scoreIsZero {
+            message.reactionScores[type.rawValue] = nil
+        }
+
+        if countIsZero {
+            message.reactionCounts[type.rawValue] = nil
+        }
+
+        if scoreIsZero && countIsZero, let reactionGroup = reactionGroup {
```

Comment:
Wondering if the check should skip `scoreIsZero` since I think I can set any kind of score value for the reaction (Int).

Reply:
I'm not sure TBH 🤔 Just to be safe, I'll leave it for now, since having the additional logic won't do warm